system_tests/bigquery.py

```diff
@@ -90,7 +92,15 @@ def test_update_dataset(self):
         after = [grant for grant in dataset.access_grants
                  if grant.entity_id != 'projectWriters']
         dataset.access_grants = after
-        dataset.update()
+
+        # We should try and keep `delay` and tries as low as possible to
+        # reduce test running time. tries=3 and delay=30 worked consistenly
+        # when updating a dataset.
```

Comment:
> Due to the nature of eventual consistency, a delay is required to wait for the dataset changes to be applied to the appropriate indexes or permissions configurations across . See: https://cloud.google.com/bigquery/streaming-data-into-bigquery#dataconsistency.
Using Google Cloud Platform Datastore can help mitigate these issues in production environments. 
See: https://cloud.google.com/datastore/docs/articles/balancing-strong-and-eventual-consistency-with-google-cloud-datastore/

Not my best writing...but something like that?

It's weird to me that permissions are giving us this trouble. The other example below makes more sense to me.